packages/hl7/src/client.ts

```diff
@@ -13,75 +12,44 @@ export class Hl7Client extends Hl7Base {
   options: Hl7ClientOptions;
   host: string;
   port: number;
-  socket?: Socket;
-  buffer: string;
-  awaitingResponse: boolean;
+  connection?: Hl7Connection;
 
   constructor(options: Hl7ClientOptions) {
     super();
     this.options = options;
     this.host = this.options.host;
     this.port = this.options.port;
-    this.buffer = '';
-    this.awaitingResponse = false;
   }
 
-  connect(): Promise<Socket> {
+  connect(): Promise<Hl7Connection> {
+    if (this.connection) {
+      return Promise.resolve(this.connection);
+    }
+
     return new Promise((resolve, reject) => {
       const socket = connect({ host: this.host, port: this.port }, () => {
-        socket.on('data', (data) => {
-          this.buffer += data.toString();
-          if (this.buffer.endsWith(FS + CR)) {
-            const msg = Hl7Message.parse(this.buffer.substring(1, this.buffer.length - 2));
-            this.dispatchEvent(new Hl7MessageEvent(socket, msg));
-            this.buffer = '';
-            this.awaitingResponse = false;
-          }
-        });
-        resolve(socket);
+        this.connection = new Hl7Connection(socket);
```

Comment:
There was a lot of duplicate code between `Hl7Server` and `Hl7Client`.  I refactored it out to `Hl7Connection`, which seems like a better abstraction, and leads to better readability (IMHO).